src/external/governance/GovernanceContract.sol

```diff
@@ -0,0 +1,339 @@
+        if (!beaconTimelock[beacon].timelockActive) {
+            revert IGovernanceContract.UpgradeProcessNotStarted();
+        }
+        _;
+    }
+
+    modifier timelockPeriodExceeded(address beacon) {
+        if (block.timestamp < beaconTimelock[beacon].timelockUntil) {
+            revert IGovernanceContract.TimelockPeriodNotExceeded();
+        }
+        _;
+    }
+
+    //--------------------------------------------------------------------------
+    // Storage
+
+    bytes32 public constant COMMUNITY_MULTISIG_ROLE = "0x01";
+    bytes32 public constant TEAM_MULTISIG_ROLE = "0x02";
+
+    address private feeManager;
+
+    uint public timelockPeriod;
+
+    mapping(address => IGovernanceContract.Timelock) private beaconTimelock;
+
+    //--------------------------------------------------------------------------
+    // Initialization
+
+    /// @notice The module's initializer function.
+    /// @param newCommunityMultisig The address of the community multisig
+    /// @param newTeamMultisig The address of the team multisig
+    /// @param newTimelockPeriod The timelock period needed to upgrade a beacon
+    function init(
+        address newCommunityMultisig,
+        address newTeamMultisig,
+        uint newTimelockPeriod //@note should we add feeManager here?
+    )
+        external
+        initializer
+        validAddress(newCommunityMultisig)
+        validAddress(newTeamMultisig)
+        validAmount(newTimelockPeriod)
+    {
+        // -> set COMMUNITY_MULTISIG_ROLE as admin of itself
+        _setRoleAdmin(COMMUNITY_MULTISIG_ROLE, COMMUNITY_MULTISIG_ROLE);
+        // -> set COMMUNITY_MULTISIG_ROLE as admin of DEFAULT_ADMIN_ROLE
+        _setRoleAdmin(DEFAULT_ADMIN_ROLE, COMMUNITY_MULTISIG_ROLE);
+
+        // Set up TEAM_MULTISIG_ROLE role structure:
+        // -> set COMMUNITY_MULTISIG_ROLE as admin of TEAM_MULTISIG_ROLE
+        _setRoleAdmin(TEAM_MULTISIG_ROLE, COMMUNITY_MULTISIG_ROLE);
+
+        // grant COMMUNITY_MULTISIG_ROLE to specified address
+        _grantRole(COMMUNITY_MULTISIG_ROLE, newCommunityMultisig);
+        // grant COMMUNITY_MULTISIG_ROLE to specified address
+        _grantRole(TEAM_MULTISIG_ROLE, newTeamMultisig);
+
+        timelockPeriod = newTimelockPeriod;
+    }
+
+    //--------------------------------------------------------------------------
+    // Getter Functions
+
+    /// @inheritdoc IGovernanceContract
+    function getBeaconTimelock(address beacon)
+        external
+        returns (Timelock memory)
+    {
+        return beaconTimelock[beacon];
+    }
+
+    //--------------------------------------------------------------------------
+    // FeeManager Functions
+
+    /// @inheritdoc IGovernanceContract
+    function getFeeManager() external view returns (address) {
+        return feeManager;
+    }
+
+    /// @inheritdoc IGovernanceContract
+    function setFeeManager(address newFeeManager)
+        external
+        onlyRole(COMMUNITY_MULTISIG_ROLE)
```

Comment:
The idea of the bigger multisig (aka community multisig) is more that it contains people that are not us or directly benefitting from the decisions like in this case. So ideally, it would, for example, be people from other projects that either advise us or are friends with us, but more importantly, do have a reputation to lose if they would just agree to negative decisions (in either direction) and want to protect our community from any harm. Does that make sense?